core/src/engine/Hamiltonian_Heisenberg.cpp

```diff
@@ -318,6 +322,10 @@ void Hamiltonian_Heisenberg::E_Anisotropy( const vectorfield & spins, scalarfiel
             if( check_atom_type( this->geometry->atom_types[ispin] ) )
                 Energy[ispin] -= this->anisotropy_magnitudes[iani]
                                  * std::pow( anisotropy_normals[iani].dot( spins[ispin] ), 2.0 );
+                Energy[ispin] -= this->cubic_anisotropy_magnitude[iani]/2 
+                                 * ( std::pow(spins[ispin][0],4.0) +
+                                     std::pow(spins[ispin][1],4.0) +
+                                     std::pow(spins[ispin][2],4.0));
```

Comment:
I'm not really sure. For our use case it's fine to put all anisotropies together.

This also depends on what are your future intentions with the anisotropy.  For us at this point the uniaxial and the 4-th order cubic is enough, but there are other terms that might be needed in some systems, such as 6-th order in-plane anisotropy for hexagonal crystals or 6-th order cubic anisotropy. If you would intend to implement other terms, it might be better to this more systematically. Right now, I just did this in the simplest way.

Reply:
I think it is slightly problematic, since it violates the assumptions of how the gradient is related to the energy. Which is used in the `Hamiltonian_Heisenberg::Gradient_and_Energy` function.
One could rewrite `Gradient_and_Energy` to not rely on these assumptions at all, which would probably be best in the long run. But of the sake of this PR I would prefer to separate it out into a cubic aniosotropy.

Reply:
The `Gradient_and_Energy` function is used to calculate the energy from the gradient-vectorfield in cases where the gradient is needed and the energy is of interest.
The expensive part of simulations tends to be the calculation of neighbours or convolutions to determine the energy-gradient, so by calculating the energy in this way, we avoid having to do the expensive calculations a second time when evaluating the energy.